### PR TITLE
Setup cd for api

### PR DESCRIPTION
**Pull request by vdhieu**
- [x] Heroku deployment for each PR (each PR has it's own blank db and url)
- [x] Auto deploy master branch master to https://spectrum-dev.herokuapp.com/

### CODE REVIEW

**Comment by vdhieu** [comment]
@all review nha ae, deploy backend api cho từng PR =))))
https://spectrum-dev-task-setup-fagt6m.herokuapp.com/graphql

File: api/tsconfig.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "module": "CommonJS",
     "moduleResolution": "node",
     "allowJs": true,
-    "noEmit": true,
+    // "noEmit": true,
```
Comment (zlatanpham) [inline api/tsconfig.json]: remove this

**Comment by vdhieu** [comment]
> Does this trigger build for any commit? If yes, can it only be triggered when changes happened in `/apis` folder?

as I know so far heroku pipeline will auto-deploy all PR, it's doesn’t care about folder changes like github action
We have 2 options 
1 > branch deploy on all PR (included changes not in /api folder)
2 > only deploy the master branch
3D > leme read the docs more, but quite sure we only have 2 options as above  =)))

**Comment by zlatanpham** [comment]
> as I know so far heroku pipeline will auto-deploy all PR, it's doesn’t care about folder changes like github action
> We have 2 options
> 1 > branch deploy on all PR (included changes not in /api folder)
> 2 > only deploy the master branch
> 3D > leme read the docs more, but quite sure we only have 2 options as above =)))

@vdhieu I think we can use `cli` for conditionally triggering the deployment - https://dev.to/heroku/deploying-to-heroku-from-github-actions-29ej. It should be similar to https://github.com/zlatanpham/spectrum-clone/blob/master/.github/workflows/storybook-pr-sync.yml.

**Pull request merged**